model_analyzer/triton/server/server_local.py

```diff
@@ -62,13 +64,11 @@ def start(self):
         if self._server_path:
             # Create command list and run subprocess
             cmd = [self._server_path]
-            cmd += self._server_config.to_cli_string().replace('=',
-                                                               ' ').split()
+            cmd += self._server_config.to_cli_string().replace('=', ' ').split()
+            # List GPUs to be used by tritonserver
             triton_env = os.environ.copy()
-            if len(self._gpus) >= 1 and self._gpus[0] != 'all':
-                visible_gpus = GPUDeviceFactory.get_cuda_visible_gpus()
-                triton_env['CUDA_VISIBLE_DEVICES'] = ','.join(
-                    [visible_gpus[uuid] for uuid in self._gpus])
+            triton_env['CUDA_VISIBLE_DEVICES'] = ','.join(
```

Comment:
Does CUDA_VISIBLE_DEVICES work with GPU UUIDs?

Reply:
It should work, because the previous code will set the `CUDA_VISIBLE_DEVICES` if the user provides UUIDs of GPU using the `gpus` config (case 1).
There are 3 cases of `gpus` config: 
   Case 1: User provides `["UUID(s) of GPU"]`
      This update does not modify the flow.
   Case 2: User provides `[]`
      This update will set the `CUDA_VISIBLE_DEVICES` to empty, while the previous code will not set it which causes a discrepancy between the GPU used by model analyzer and triton server.
   Case 3: User provides `all`
      This update will set the `CUDA_VISIBLE_DEVICES` to includes all GPU UUIDs visible by CUDA, while the previous code will not set it. This should not change the behavior.